src/ValidationFailed.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Libero\XmlValidator;
+
+use Throwable;
+use UnexpectedValueException;
+
+class ValidationFailed extends UnexpectedValueException
+{
+    private $failures;
+
+    /**
+     * @param ValidationFailure[] $failures
+     */
+    public function __construct(array $failures, string $message = '', int $code = 0, ?Throwable $previous = null)
```

Comment:
what type `$failures` are is a bit underspecified. Strings? Objects?

Reply:
Ah, `ValidationFailure` objects, possible overlap of names as they are very similar

Reply:
Could probably be just `Failure` given the namespace, to clearly distinguish exception and its contents?

Reply:
Yeah, the failure isn't necessarily a validation one too (eg could be the schema doesn't exist).

Reply:
See also https://github.com/libero/xml-validator/pull/1#discussion_r223354063.